src/mmrelay/meshtastic/daemon_executor.py

```
@@ -0,0 +1,109 @@
+
+    ``ThreadPoolExecutor`` deliberately uses non-daemon workers and CPython waits
+    for those workers during interpreter shutdown. That is desirable for normal
+    jobs, but not for best-effort BLE calls into BlueZ/DBus that can wedge below
+    Python and never return. This executor keeps the familiar ``Future`` API while
+    ensuring a stuck worker cannot prevent the relay process from exiting.
+
+    The implementation intentionally uses only public ``concurrent.futures``
+    primitives so it does not depend on private CPython thread-pool internals.
+    """
+
+    def __init__(self, max_workers: int = 1, *, thread_name_prefix: str = "daemon"):
```

Ruff (0.16.0)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)